### DISCUSSION TITLE
What are 'base' classes for in effects?

### DISCUSSION THREAD

**Discussion by weow147** [pschatzmann/arduino-audio-tools#1249] [Q&A] [answered]
I don't conceptually understand the base classes:

- [EffectSuitBase](https://pschatzmann.github.io/arduino-audio-tools/classaudio__tools_1_1_effect_suite_base.html)
- [STKEffect](https://pschatzmann.github.io/arduino-audio-tools/classaudio__tools_1_1_s_t_k_effect.html)

Could somebody describe a use case?  Are they just containers for Arduino Audio Tools or STK Framework effects?

In any case, I can't figure out how to implement them?

**Comment by pschatzmann** [comment, marked as answer]
STKEffect is just an adaper class that lets you use the effects from the [STK framework](https://github.com/pschatzmann/arduino-stk). You would just provide a [stk effect](https://pschatzmann.github.io/arduino-stk/html/classstk_1_1Effect.html) in the constructor. Alternatively you can directly use the [effect implementations that start with STK](https://pschatzmann.github.io/arduino-audio-tools/classaudio__tools_1_1_audio_effect.html)

The effectsuit is just an adaped effect implementation from mhamilt. The EffectSuiteBase is just an abstract base class and you would use the dependent effect classes (e.g SimpleChorus)

**Reply by weow147** [reply]
I will try to use it to deconflict 'ambiguous references' errors i'm getting when trying to use STK and 'echo' or 'delay' at the same time.

**Comment by pschatzmann** [comment]
I don't understand. Can you provide an example ?
